### PR TITLE
Add script for sending release message to Discord

### LINKED ISSUE #12354: Send Discord release message through GitHub Actions
Our current release message on Discord uses webhooks. We can use GitHub Actions instead for a better format.

FYI: https://github.com/marketplace/actions/send-discord-message

The `send-discord-message` is just a random search I found, we can look through it to see if there are more popular ones or write our own script.

### PR DESCRIPTION
**Pull request by LitoMore**
Resolves #12354

### How to preview

1. Create a private channel for testing
2. Create a webhook bot that belongs to your channel
3. Create a role for testing
4. Save webhook link and role link to your local environment variable:
    ```shell
    export DISCORD_RELEASES_WEBHOOK_URL="https://discord.com/api/webhooks/..."
    export DISCORD_RELEASES_ROLE_ID="123456"
    ```
5. Run `node scripts/release/discord-release-message.js 1.2.3` to preview

### CODE REVIEW

**Review by PratyushJayachandran — APPROVED** [review]

**Review by mondeja — COMMENTED** [review]
It seems to me like a bit overkilling for just an HTTP request. Why you didn't used cURL directly?

**Comment by LitoMore** [comment]
With JavaScript we can build the release message (links, mentions) much easier.

I have a plan to add a more detailed content (PR link, contributors) to the message.

File: scripts/release/discord-release-message.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    method: 'post',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      content: `<@&${discordReleasesRoleId}> ${githubReleaseUrl}`,
+    }),
+  });
+} catch {
+  console.error('Failed to send release message to Discord.');
```
Comment (mondeja) [inline scripts/release/discord-release-message.js:23]:
```suggestion
  console.error('Failed to send release message to Discord.');
  process.exit(1);
```

Comment (LitoMore) [reply scripts/release/discord-release-message.js:23]: I intentionally did not exit with code 1 because I don't want to block the rest of the steps.

Comment (mondeja) [reply scripts/release/discord-release-message.js:23]: Use [`continue-on-error`](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error) on the GHA step.

**Review by mondeja — APPROVED** [review]

**Pull request merged**